docs/user-guide/rug/command-handlers.md

```diff
@@ -0,0 +1,239 @@
+of the `#!typescript @CommandHandler` decorator is the name of the command,
+the second is its description. These make the handlers visible and discoverable.
+
+### Discovery
+
+{!decorators/tags.md!}
+
+[decorators]: https://www.typescriptlang.org/docs/handbook/decorators.html
+
+### Implementation
+
+We then define the class which implements our command handler (line 9). It is
+exported so that it can referenced from any unit tests. A command handler
+implements the `#!typescript HandleCommand` interface. This interface requires
+the `#!typescript handle(command: HandlerContext): CommandPlan` method to be 
+implemented. It is a convention for the command handler and the class that defines
+it to have the same name.
+
+The `#!typescript handle` method takes a single argument, a
+`#!typescript HandlerContext` instance. This gives you access to a
+[path expression engine][pxe] to query your organization's projects. The method
+must return a `#!typescript CommandPlan`.
+
+[pxe]: path-expressions.md
+
+### Intent
+
+A Rug command handler can have associated intent users can send when talking
+with the Atomist bot. The intent is described using the `#!typescript @Intent`
+decorator (line 8). Whenever a user sends the `@atomist release changelog` message
+to the Atomist bot, the Rug runtime runs the `#!typescript UpdateChangeLog`
+command handler.
+
+### Parameters
+
+Rug command handlers can take parameters like other Rugs. {!decorators/parameters.md!}
+
+[rugconv]: conventions.md
+
+### Mapped Parameters
+
+Rug command handlers can define what are called Mapped Parameters to receive
+relevant contextual information when invoked via the Atomist bot.
+
+{!decorators/mapped-parameters.md!}
+
+### CommandPlans
+
+An `#!typescript CommandPlan` describes the actions to be taken by the Rug runtime
+on behalf of the handler. CommandPlans are composed of [Messages](#messages) and/or 
+[respondables](#respondables). Respondables instruct the rug runtime to immediately
+perform ordinary rug operations, whereas messages are sent to the Atomist Bot for
+display to the user.
+
+### Messages
+
+A `Message` represents presentable content to be rendered in the chat system by
+the Atomist Bot to a chat channel or user. However, each of the two available
+message types achieve this in different ways.
+
+#### Response Messages
+
+{!response-message.md!}
+
+#### Directed Messages
+
+{!directed-message.md!}
+
+### Respondables
+
+An `#!typescript CommandRespondable` is really just a container for an instruction and
+some optional `onError` and `onSuccess` capabilities. The `onError` and `onSuccess`
+properties of an `#!typescript CommandRespondable` can be [messages](#messages), 
+[CommandPlans](#eventplans) or [response handlers](#response-handlers).
+
+```typescript
+let plan = new CommandPlan();
+plan.add(
+    { 
+        instruction: { 
+            kind: "execute", 
+            name: "create-github-issue",
```

Comment:
I don't think so as this is the name of the function, which I _think_ is kebab case. :)